### PR TITLE
"Feature #148311 feat: Ordering for Option in fields (radio, single, multi select)

### PR DESCRIPTION
**Pull request by shraddhaSankpal27**

### CODE REVIEW

File: administrator/models/field.php [hunk 1]
```diff
@@ -139,14 +139,20 @@ public function getItem($pk = null)
...
+				$query->select('opt.id as optionid,opt.options as name,opt.value FROM #__tjfields_options as opt');
 				$query->where('opt.field_id=' . $input->get('id', '', 'INT'));
+				$query->order('opt.ordering ASC');
 				$db->setQuery($query);
 				$option_name = $db->loadObjectlist();
 
 				if ($option_name)
 				{
-					$item->fieldoption = $option_name;
+					foreach ($option_name as $key => $option)
```
Comment (ankush-maherwal) [inline administrator/models/field.php]: @shraddhaSankpal27 Why do we need this processing?

We will be breaking the method contract by changing the output of the method.

Cant we avoid this?

File: administrator/models/field.php [hunk 2]
```diff
@@ -360,16 +366,16 @@ public function save($data)
...
 					// Check for empty options
-					if ((!empty($value['optionname']) && $value['optionvalue'] == '') || ($value['optionname'] == '' && !empty($value['optionvalue'])))
+					if ((!empty($value['name']) && $value['value'] == '') || ($value['name'] == '' && !empty($value['value'])))
 					{
 						JFactory::getApplication()->enqueueMessage(JText::_('COM_TJFIELDS_INVALID_OPTION_VALUES'), 'error');
 
 						return $id;
 					}
 
-					if ($value['hiddenoptionid'])
```
Comment (ankush-maherwal) [inline administrator/models/field.php]: @shraddhaSankpal27 why the 'hiddenoptionid' is changed to 'optionid'?

Have you checked the full impact? Seems that the 'hiddenoptionid' is used in some JS too

File: administrator/models/forms/field.xml [hunk 3]
```diff
@@ -58,7 +57,40 @@
 			<option value="0">COM_TJFIELDS_NO</option>
 		</field>
 	</fieldset>
-
+	<fields name="fieldparams">
```
Comment (ankush-maherwal) [inline administrator/models/forms/field.xml]: @shraddhaSankpal27 why do we need to define separate fieldset for options filed?

We can add that in the above fieldset only

File: administrator/models/field.php (hunk 2 above)
Comment (shraddhaSankpal27) [reply administrator/models/field.php]: @ankush-maherwal hiddenoptionid is used in field.js files addclone(), but addClone method is used in fieldOption.php, as we discussed we are removing the fieldOption.php layout.Hence it is not impacted on any other code.

File: administrator/models/field.php (hunk 1 above)
Comment (shraddhaSankpal27) [reply administrator/models/field.php]: @ankush-maherwal Done

File: administrator/models/forms/field.xml (hunk 3 above)
Comment (shraddhaSankpal27) [reply administrator/models/forms/field.xml]: @ankush-maherwal removed fieldset

File: administrator/models/field.php [hunk 4]
```diff
@@ -139,15 +145,13 @@ public function getItem($pk = null)
 			{
 				$db = JFactory::getDbo();
 				$query = $db->getQuery(true);
-				$query->select('opt.id,opt.options,opt.value FROM #__tjfields_options as opt');
+				$query->select('opt.id as optionid,opt.options as name,opt.value FROM #__tjfields_options as opt');
```
Comment (ankush-maherwal) [inline administrator/models/field.php]: @shraddhaSankpal27 Have you checked the impact of changing the names `opt.id` => `optionid` and `opt.options` => `name` in all the places where it is used

**Pull request merged**